homeassistant/components/bluetooth/manager.py

```diff
@@ -307,24 +272,25 @@ def _async_all_discovered_addresses(self, connectable: bool) -> Iterable[str]:
                 for scanner in self._non_connectable_scanners
             )
 
-    @hass_callback
     def async_discovered_devices(self, connectable: bool) -> list[BLEDevice]:
         """Return all of combined best path to discovered from all the scanners."""
         histories = self._connectable_history if connectable else self._all_history
         return [history.device for history in histories.values()]
 
-    @hass_callback
     def async_setup_unavailable_tracking(self) -> None:
         """Set up the unavailable tracking."""
-        self._cancel_unavailable_tracking = async_track_time_interval(
-            self.hass,
-            self._async_check_unavailable,
-            timedelta(seconds=UNAVAILABLE_TRACK_SECONDS),
-            name="Bluetooth manager unavailable tracking",
+        self._schedule_unavailable_tracking()
+
+    def _schedule_unavailable_tracking(self) -> None:
+        """Schedule the unavailable tracking."""
+        if TYPE_CHECKING:
+            assert self._loop is not None
+        loop = self._loop
+        self._cancel_unavailable_tracking = loop.call_at(
+            loop.time() + UNAVAILABLE_TRACK_SECONDS, self._async_check_unavailable
         )
```

Comment:
Since this is being extracted we need to switch to loop scheduling instead of Home Assistant internals